Source/WebCore/platform/soup/PublicSuffixSoup.cpp

```diff
@@ -67,7 +86,14 @@ String topPrivatelyControlledDomain(const String& domain)
     if (const char* baseDomain = soup_tld_get_base_domain(domainUTF8.data() + position, &error.outPtr()))
         return String::fromUTF8(baseDomain);
 
-    if (g_error_matches(error.get(), SOUP_TLD_ERROR, SOUP_TLD_ERROR_INVALID_HOSTNAME) || g_error_matches(error.get(), SOUP_TLD_ERROR, SOUP_TLD_ERROR_NOT_ENOUGH_DOMAINS) || g_error_matches(error.get(), SOUP_TLD_ERROR, SOUP_TLD_ERROR_NO_BASE_DOMAIN))
+    if (g_error_matches(error.get(), SOUP_TLD_ERROR, SOUP_TLD_ERROR_NO_BASE_DOMAIN)) {
+        // We manually handle web-platform.test used for WPT tests as it is not a real TLD.
+        if (domain.endsWithIgnoringASCIICase("web-platform.test"_s))
```

Comment:
I am not sure it is great to have this check here.
Maybe we should do like for scheme registries and have a list of specific domains for which we would use this heuristic.
Then our test tools would add "web-platform.test" to this list.

Reply:
This domain will be found in `LegacyGlobalSettings::hostnamesToRegisterAsLocal()`, should we re-use this?

Reply:
Seems god

Reply:
Unfortunately `LegacyGlobalSettings` is in the wrong layer and WebCore doesn't have access to it.